internal/parser/scanner/scanner_test.go

```diff
@@ -65,6 +57,120 @@ func Test_randomTokenSequence(t *testing.T) {
 	}
 }
 
+func Test_scanOperator(t *testing.T) {
+	desiredOutput := []string{
+		"||",
+		"*",
+		"/",
+		"%",
+		"+",
+		"-",
+		"<<",
+		">>",
+		"&",
+		"|",
+		"<",
+		"<=",
+		">=",
+		"=",
+		"==",
+		"!=",
+		"<>",
+		"~",
+	}
+	desiredOutputType := []token.Type{
+		token.BinaryOperator,
+		token.BinaryOperator,
+		token.BinaryOperator,
+		token.BinaryOperator,
+		token.UnaryOperator,
+		token.UnaryOperator,
+		token.BinaryOperator,
+		token.BinaryOperator,
+		token.BinaryOperator,
+		token.BinaryOperator,
+		token.BinaryOperator,
+		token.BinaryOperator,
+		token.BinaryOperator,
+		token.BinaryOperator,
+		token.BinaryOperator,
+		token.BinaryOperator,
+		token.BinaryOperator,
+		token.UnaryOperator,
+	}
+
+	var scannerOutput []token.Type
+	for i := range desiredOutput {
+		scanner := New([]rune(desiredOutput[i]))
+		for scanner.HasNext() {
+			nextToken := scanner.Next()
+			if nextToken.Type() == token.Error {
+				t.Errorf("Error token recieved - %v\n", nextToken.Value())
+			} else {
+				scannerOutput = append(scannerOutput, nextToken.Type())
+			}
+		}
+	}
+
+	for i, op := range desiredOutputType {
+		if op != scannerOutput[i] {
+			t.Errorf("Expected %v, obtained %v\n", op, scannerOutput[i])
+		}
+	}
+
+	if t.Failed() {
+		fmt.Printf("Expected %v, obtained %v\n", desiredOutputType, scannerOutput)
+	} else {
+		fmt.Printf("Tested on: %v\n", desiredOutput)
+	}
+}
+
+func Test_multipleSequences(t *testing.T) {
+	inputSequences := []string{
+		"SELECT    *      FROM      users",
+		"SELECT       FROM || & +7 59 \"foobar\"",
+	}
+	desiredOutput := [][]string{
+		[]string{
+			"SELECT",
+			"*",
+			"FROM",
+			"users",
+		},
+		[]string{
+			"SELECT",
+			"FROM",
+			"||",
+			"&",
+			"+",
+			"7",
+			"59",
+			"\"foobar\"",
+		},
+	}
+	desiredOutputType := [][]token.Type{
+		[]token.Type{
+			token.KeywordSelect,
+			token.BinaryOperator,
+			token.KeywordFrom,
+			token.Literal,
+		},
+		[]token.Type{
+			token.KeywordSelect,
+			token.KeywordFrom,
+			token.BinaryOperator,
+			token.BinaryOperator,
+			token.UnaryOperator,
+			token.Literal,
+			token.Literal,
+			token.Literal,
+		},
+	}
```

Comment:
You should also test the position information of the tokens.

Reply:
Since the test happens as an iteration, I think order is maintained isnt it?

Reply:
Done.